app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -0,0 +1,29 @@
+module HasMonitoringManagerMixin
+  extend ActiveSupport::Concern
+
+  included do
+    # TODO: why is this relation needed ? why is the inheritors relation no enough?
+    has_one :monitoring_manager,
+            :foreign_key => :parent_ems_id,
+            :class_name  => "ManageIQ::Providers::NetworkManager",
+            :autosave    => true,
+            :dependent   => :destroy
+
+    private
+
+    def ensure_monitoring_manager
+      if endpoints.size > 1
```

Comment:
Right, I also had a version checking for 'prometheus' | 'hawkular' roles but that felt too specific for this level.
Maybe the best solution is to offload this to the implementations?